### PR TITLE
Added initial IIB flows

### PR DESCRIPTION
**Pull request by ndegreef**
Added initialversion of IIB provisioning flows

### CODE REVIEW

File: provisioning-workflows/IIB/IIBv9/jcl/AUTHFS.jcl [hunk 1]
```diff
@@ -0,0 +1,43 @@
+//S1     EXEC USSBATCH,SU=Y
```
Comment (RonaldKristel) [inline provisioning-workflows/IIB/IIBv9/jcl/AUTHFS.jcl:1]: USSBATCH is not a standard utility available on z/OS, maybe we should find a more standard approach.

File: provisioning-workflows/IIB/IIBv9/jcl/AUTHFS.jcl [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+# gaat recursief door directories vanaf startpunt, slaat links over
+# reageert ook goed op spaties in directory namen
+function dodir
+{
+        typeset fname savedir
+        ls | while read "fname" ; do
+                if [[ -L "$fname" ]] ; then
+                        echo 'skipped link '  "$fname"
+                elif [[ -f "$fname" ]] ; then
+                        setfacl -m group:USNZOS01:rwx "$fname"
```
Comment (RonaldKristel) [inline provisioning-workflows/IIB/IIBv9/jcl/AUTHFS.jcl]: Hardcoded SAF group.

File: provisioning-workflows/IIB/IIBv9/jcl/AUTHFS.jcl [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                        setfacl -m user:${instance-ZOSMF_INSTALL_USER}:rwx "$fname"
+
+                elif [[ -d "$fname" ]] ; then
+                        savedir=$(pwd)
+                        setfacl -m group:USNZOS01:rwx "$fname"
```
Comment (RonaldKristel) [inline provisioning-workflows/IIB/IIBv9/jcl/AUTHFS.jcl]: Hardcoded SAF group

File: provisioning-workflows/IIB/IIBv9/jcl/AUTHFS.jcl [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+                        setfacl -m user:${instance-ZOSMF_INSTALL_USER}:rwx "$fname"
+                        setfacl -m f:group:USNZOS01:rwx   ./
```
Comment (RonaldKristel) [inline provisioning-workflows/IIB/IIBv9/jcl/AUTHFS.jcl]: Hardcoded SAF group

File: provisioning-workflows/IIB/IIBv9/jcl/AUTHFS.jcl [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+cd $fname2 && \
+dodir $(pwd)
+setfacl -m group:USNZOS01:rwx ./           
+setfacl -m d:group:USNZOS01:rwx   ./       
+
+setfacl -m user:${instance-ZOSMF_INSTALL_USER}:rwx ./           
+setfacl -m d:user:${instance-ZOSMF_INSTALL_USER}:rwx   ./
+
+exit 0
+
```
Comment (RonaldKristel) [inline provisioning-workflows/IIB/IIBv9/jcl/AUTHFS.jcl:43]: Line 15, 20, 22, 36, 37  all reference hardcoded RACF groups. We should create workflow variable for that.

File: provisioning-workflows/IIB/IIBv9/jcl/CRENWDR.jcl [hunk 6]
```diff
@@ -0,0 +1,10 @@
+//S1     EXEC USSBATCH,SU=Y
```
Comment (RonaldKristel) [inline provisioning-workflows/IIB/IIBv9/jcl/CRENWDR.jcl:1]: USSBATCH is not a standard utility available on z/OS, maybe we should find a more standard approach.

File: provisioning-workflows/IIB/IIBv9/jcl/DeleteMountpointAndBrokerFS.jcl [hunk 7]
```diff
@@ -0,0 +1,15 @@
+//MOUNT    EXEC PGM=IKJEFT01
+//SYSTSPRT DD SYSOUT=*
+//SYSTSIN  DD *
+UNMOUNT FILESYSTEM('SYSTEM.HFS.PLEX.VAR.MQSI.${instance-IIB_BROKER_NAME}') +
+IMMEDIATE
+/*
+//S1     EXEC USSBATCH,SU=Y
```
Comment (RonaldKristel) [inline provisioning-workflows/IIB/IIBv9/jcl/DeleteMountpointAndBrokerFS.jcl:7]: USSBATCH is not a standard utility available on z/OS, maybe we should find a more standard approach.

File: provisioning-workflows/IIB/IIBv9/jcl/EnvironmentFileTemplate.txt [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+#
+# 6.9 MQSI_FAD current FAD level.
+#
+export _BPX_BATCH_SPAWN=NO
+export MQSI_MC_MESSAGES=NO
+export MQSI_COMMAND_DATABASE_ECHO=1
+export MQSI_COMMAND_ZOS_MQ_ECHO=1
+export MQSI_COMMAND_ZOS_MQ_ECHO_RC=1
+export MQSI_DEPLOY_PROGRESS=1
+export STEPLIB=SYS3.${instance-IIB_DBSRV_NAME}.SDSNEXIT:SYS3.${instance-IIB_DBSRV_NAME}.SDSNLOAD:SYS3.${instance-IIB_DBSRV_NAME}.SDSNLOD2
```
Comment (RonaldKristel) [inline provisioning-workflows/IIB/IIBv9/jcl/EnvironmentFileTemplate.txt]: SYS3 is assumed here, but should probably made variable. Just like &DB2HLQ in /jcl/UPDSTC.jcl ?

File: provisioning-workflows/IIB/IIBv9/provisionIIB.properties [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+IIB_MQ_PATH_new = /usr/lpp/mqm/V9R0M0
+IIB_JAVAHOME = /usr/lpp/java/J7.1_64
+IIB_MQ_PATH = /usr/lpp/mqm/V9R0M0
+IIB_USS_BIN = /usr/lpp/mqsi/V9R0M0/bin
+IIB_STCDFLTGROUP = USSDBMT5
+IIB_STCGROUP = FUSTCBRK
+IIB_JDBC_JARS = /usr/lpp/db2b10/jdbc/classes
+#IIB_PRD_USER = PT5S
+IIB_PRD_USER = PT5X
+IIB_TIV_USER = TIVMANU
```
Comment (RonaldKristel) [inline provisioning-workflows/IIB/IIBv9/provisionIIB.properties:31]: We should include some comments here, and remove all references to site specific settings; much like how sample files today are provided:

> (e.g.)
>`## The "IIB_MQ_DS_PREFIX" parameter indentifies the High Level Qualifier for the MQ Series datasets. 
>IIB_MQ_DS_PREFIX = SYS3

>`## The "ZOSMF_PARMLIB" parameter indentifies which Parmlib member is used.
>ZOSMF_PARMLIB = SYS1.PARMLIB

File: provisioning-workflows/IIB/IIBv9/provisionIIB.xml [hunk 10]
```diff
@@ -0,0 +1,1055 @@
...
+//SYSPRINT DD SYSOUT=*
+//SYSTSIN DD *
+
+PE ${instance-ZOSMF_INSTANCE}.ZOSMF.TEMPLATE.RUNASUSERS.${_workflow-domainID}.${_workflow-templateName} - 
+ CLASS(ZMFCLOUD) ID($IIB_STCUSER) ACCESS(READ)
+PE ${MQCONNBATCH}.BATCH CL(MQCONN) ID(${instance-ZOSMF_INSTALL_USER}) ACCESS(READ)
+PE *.DELETE.QLOCAL CL(MQCMDS) ID($IIB_STCUSER) ACCESS(ALTER)
+
+PE IEFUSI.PLUS256 CLASS(FACILITY) ID($IIB_STCUSER) ACCESS(READ)
+PE IEFUSI.PLUS256 CLASS(FACILITY) ID(${instance-ZOSMF_INSTALL_USER}) ACCESS(READ)
```
Comment (RonaldKristel) [inline provisioning-workflows/IIB/IIBv9/provisionIIB.xml]: IEFUSI.PLUS256 is site specific, remove.

File: provisioning-workflows/IIB/IIBv9/jcl/deriveVariables.jcl [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+ when phase = "P" then do
+            db2subsys = "DBEF"
+            db2port = 1446
+          end
+ otherwise do
+            db2subsys = "UNKNOWN"
+            db2port = "ERROR"
+            Say "DB2 subsystem for environment could not be determined"
+            Exit 999
+          end
```
Comment (RonaldKristel) [inline provisioning-workflows/IIB/IIBv9/jcl/deriveVariables.jcl:44]: Site specific DB2 select,

**Review by RonaldKristel — COMMENTED** [review]
There is some site specific code in here that will mostly likely not work if this is provided as sample to the community. Some code needs to be changed in order to work for a broader audience.  

What is the common approach here? Do we approve changes and work from there?

File: provisioning-workflows/IIB/IIBv9/jcl/AUTHFS.jcl (hunk 1 above)
Comment (shreeji818) [reply provisioning-workflows/IIB/IIBv9/jcl/AUTHFS.jcl:1]: I agree. Do we have to wait for Niek to make change or @RonaldKristel will you be able to make this change?

File: provisioning-workflows/IIB/IIBv9/jcl/AUTHFS.jcl (hunk 5 above)
Comment (shreeji818) [reply provisioning-workflows/IIB/IIBv9/jcl/AUTHFS.jcl:43]: I agree. Do we have to wait for Niek to make change or @RonaldKristel will you be able to make this change?

**Comment by shreeji818** [comment]
I agree with the comments. Do we have to wait for Niek to make change or @RonaldKristel will you be able to make these changes?

Normally we should not have any site specific code. We can either use variables that would make it non site-specific or use "instruction" if there is some site specific code required with example in instructional step.

File: provisioning-workflows/IIB/IIBv9/jcl/CRENWDR.jcl (hunk 6 above)
Comment (RonaldKristel) [reply provisioning-workflows/IIB/IIBv9/jcl/CRENWDR.jcl:1]: USSBATCH is now supplied with workflow.

File: provisioning-workflows/IIB/IIBv9/jcl/AUTHFS.jcl (hunk 1 above)
Comment (RonaldKristel) [reply provisioning-workflows/IIB/IIBv9/jcl/AUTHFS.jcl:1]: USSBATCH is now supplied as seperate artifact in workflow folder.

**Review by RonaldKristel — APPROVED** [review]

**Pull request merged**